internal/services/appconfiguration/app_configuration_data_source.go

```diff
@@ -44,6 +44,11 @@ func dataSourceAppConfiguration() *pluginsdk.Resource {
 				Computed: true,
 			},
 
+			"public_network_access": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
```

Comment:
as this is essential a bool could we make it
```suggestion
			"public_network_access_enabled": {
				Type:     pluginsdk.TypeBool,
				Computed: true,
			},
```

this would also be consistant with the rest of the provider

Reply:
Chatting about this internally this is fine to leave as the string values here - the `Public Network Access` fields are being changed from a two-value constant (which we expose as a boolean, due to the values being Enabled/Disabled) - to a three-value constant. At some point to support this new third value we're going to need to migrate from using a Boolean for this to using a Constant for all resources which currently support this - but that'd be a breaking change and so isn't something we currently plan to do until v4.0.

Since this is a new data source/resource we should look to expose the constants for this - so that we don't need to introduce a breaking change later. Whilst this isn't ideal from a UX perspective (as we'd rather they were consistent) - the alternative would be a breaking change which would cause other issues - so in short, this is fine to leave as-is 👍

Reply:
Thanks (to both). Let me know if there is anything else that I could be missing.